packages/next/src/build/flying-shuttle/store-shuttle.ts

```diff
@@ -8,20 +12,57 @@ import {
   PAGES_MANIFEST,
   ROUTES_MANIFEST,
 } from '../../shared/lib/constants'
-import { recursiveCopy } from '../../lib/recursive-copy'
+
+export function getShuttleManifest(config: NextConfigComplete) {
+  const globalHash = crypto.createHash('sha256')
+
+  for (const key of Object.keys(process.env || {})) {
+    if (key.startsWith('NEXT_PUBLIC_')) {
+      globalHash.update(`${key}=${process.env[key]}`)
+    }
+  }
+
+  const omittedConfigKeys = ['headers', 'rewrites', 'redirects']
+
+  for (const key of Object.keys(config)) {
+    if (omittedConfigKeys.includes(key)) {
+      continue
+    }
+    let serializedConfig =
+      typeof config[key] === 'function'
+        ? config[key].toString()
+        : JSON.stringify(config[key])
+
+    globalHash.update(`${key}=${serializedConfig}`)
+  }
+
+  return {
+    nextVersion,
+    // NEXT_PUBLIC_ changes for now and specific next config values
```

Comment:
should this comment be earlier in the function where we're determining which keys to process?